Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/HomeFragmentViewModel.kt

```diff
@@ -0,0 +1,69 @@
+package de.rki.coronawarnapp.ui.main.home
+
+import com.squareup.inject.assisted.AssistedInject
+import de.rki.coronawarnapp.risk.TimeVariables
+import de.rki.coronawarnapp.storage.LocalData
+import de.rki.coronawarnapp.timer.TimerHelper
+import de.rki.coronawarnapp.ui.viewmodel.SettingsViewModel
+import de.rki.coronawarnapp.ui.viewmodel.SubmissionViewModel
+import de.rki.coronawarnapp.ui.viewmodel.TracingViewModel
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.security.EncryptionErrorResetTool
+import de.rki.coronawarnapp.util.ui.SingleLiveEvent
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
+
+class HomeFragmentViewModel @AssistedInject constructor(
+    dispatcherProvider: DispatcherProvider,
+    private val errorResetTool: EncryptionErrorResetTool,
+    val tracingViewModel: TracingViewModel,
+    val settingsViewModel: SettingsViewModel,
+    val submissionViewModel: SubmissionViewModel
+) : CWAViewModel(
+    dispatcherProvider = dispatcherProvider,
+    childViewModels = listOf(tracingViewModel, settingsViewModel, submissionViewModel)
+) {
+
+    val showInteropDeltaOnboarding = SingleLiveEvent<Unit>()
+    val showTracingExplanation = SingleLiveEvent<Long>()
+    val showErrorResetDialog = SingleLiveEvent<Boolean>()
```

Comment:
sealed event class?

Reply:
Note sure what you mean. Something like `data class Event<T>(val value: T)`? Don't think we need that here.

Reply:
like SymptomCalendarEvent - so that we need only 1 life data for the event - as agreed on at https://teams.microsoft.com/l/entity/com.microsoft.teamspace.tab.wiki/tab::32332847-46cb-4995-a3eb-e183b8c68f28?context=%7B%22subEntityId%22%3A%22%7B%5C%22pageId%5C%22%3A21%2C%5C%22sectionId%5C%22%3A22%2C%5C%22origin%5C%22%3A2%7D%22%2C%22channelId%22%3A%2219%3Ac51bf302889a44e28119b0efc59b8f7d%40thread.tacv2%22%7D&tenantId=42f7676c-f455-423c-82f6-dc2d99791af7

Reply:
:sweat_smile:  you are absolutely right, changed. Looks much better now :+1: